app/org/sagebionetworks/bridge/play/controllers/StudyConsentController.java

```diff
@@ -21,54 +24,132 @@
 public class StudyConsentController extends BaseController {
 
     private StudyConsentService studyConsentService;
+    
+    private SubpopulationService subpopService;
 
     @Autowired
-    public void setStudyConsentService(StudyConsentService studyConsentService) {
+    final void setStudyConsentService(StudyConsentService studyConsentService) {
         this.studyConsentService = studyConsentService;
     }
 
+    @Autowired
+    final void setSubpopulationService(SubpopulationService subpopService) {
+        this.subpopService = subpopService;
+    }
+
+    // V1 API: consents directly associated to a study
+    
+    @Deprecated
     public Result getAllConsents() throws Exception {
         UserSession session = getAuthenticatedSession(RESEARCHER);
+        return getAllConsentsV2(session.getStudyIdentifier().getIdentifier());
+    }
+
+    @Deprecated
+    public Result getActiveConsent() throws Exception {
+        UserSession session = getAuthenticatedSession(RESEARCHER);
+        return getActiveConsentV2(session.getStudyIdentifier().getIdentifier());
+    }
+    
+    @Deprecated
+    public Result getMostRecentConsent() throws Exception {
+        UserSession session = getAuthenticatedSession(RESEARCHER);
+        return getMostRecentConsentV2(session.getStudyIdentifier().getIdentifier());
+    }
+
+    @Deprecated
+    public Result getConsent(String createdOn) throws Exception {
+        UserSession session = getAuthenticatedSession(RESEARCHER);
+        return getConsentV2(session.getStudyIdentifier().getIdentifier(), createdOn);
+    }
+    
+    @Deprecated
+    public Result addConsent() throws Exception {
+        UserSession session = getAuthenticatedSession(RESEARCHER);
+        return addConsentV2(session.getStudyIdentifier().getIdentifier());
+    }
+
+    @Deprecated
+    public Result publishConsent(String createdOn) throws Exception {
+        UserSession session = getAuthenticatedSession(RESEARCHER);
+        return publishConsentV2(session.getStudyIdentifier().getIdentifier(), createdOn);
+    }
+    
+    // V2: consents associated to a subpopulation
```

Comment:
Added tests for all the methods that should accept a guid and pass it property to the services.